app/src/hooks/useAppLanguage.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { getAppLanguage } from "store/selectors";
+import { globalActions } from "store/slices/global/slice";
+
+export const useAppLanguage = () => {
+  const dispatch = useDispatch();
+  const currentLanguage = useSelector(getAppLanguage);
+
+  useEffect(() => {
+    const observer = new MutationObserver(() => {
+      const newLang = document.documentElement.lang;
+      if (newLang !== currentLanguage) {
```

Comment:
No need of this check, if `newLang === currentLanguage` there will be no update after the dispatch, since the state is same, this will also remove the `currentLanguage` from dependency array.